File: Task3/Ruby/subtask4.rb
lines = File.readlines('input.txt')
n_line = lines.find { |line| line.start_with?('n') }
n = n_line.split('=').last.to_i
n += 1 if n.even?

pattern = ""

(1..n).step(2) { |i| pattern << "#{' ' * ((n - i) / 2)}#{'*' * i}\n" }
(n - 2).step(1, -2) { |i| pattern << "#{' ' * ((n - i) / 2)}#{'*' * i}\n" }

File.write('output.txt', pattern)

File: Task3/Python/subtask4.py
with open('input.txt', 'r') as file:
    n_line = next(line for line in file if line.startswith('n'))
n = int(n_line.split('=')[1].strip())
if n % 2 == 0:
    n += 1

pattern = ""

for i in range(1, n + 1, 2):
    pattern += ' ' * ((n - i) // 2) + '*' * i + '\n'
for i in range(n - 2, 0, -2):
    pattern += ' ' * ((n - i) // 2) + '*' * i + '\n'

with open('output.txt', 'w') as file:
    file.write(pattern)
